### PR TITLE
Add: Steam API tool

### PR DESCRIPTION
**Pull request by muntaqamahmood**
- **Description:** Our PR is an integration of a Steam API Tool that makes recommendations on steam games based on user's Steam profile and provides information on games based on user provided queries.
  - **Issue:** the issue # our PR implements: https://github.com/langchain-ai/langchain/issues/12120
  - **Dependencies:** python-steam-api library, steamspypi library and decouple library
  - **Tag maintainer:** @baskaryan, @hwchase17 
  - **Twitter handle:** N/A

Hello langchain Maintainers,

We have run make format, make lint and make test locally before submitting the PR. To our knowledge, our changes do not introduce any new errors.

Our PR integrates the python-steam-api, steamspypi and decouple packages. We have added integration tests to test our python API integration into langchain and an example notebook is also provided.

Our amazing team that contributed to this PR: @JohnY2002, @shenceyang, @andrewqian2001 and @muntaqamahmood  

Thank you in advance to all the maintainers for reviewing our PR!

### CODE REVIEW

File: libs/langchain/langchain/utilities/steam.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
+"""Util that calls Steam-WebAPI."""
+
+from typing import Any
+
+import steamspypi
```
Comment (hwchase17) [inline libs/langchain/langchain/utilities/steam.py:5]: these should be conditionally imported

**Review by hwchase17 — COMMENTED** [review]
the notebook needs a title, and theres also an error in it currently

File: libs/langchain/langchain/utilities/steam.py (hunk 1 above)
Comment (andrewqian2001) [reply libs/langchain/langchain/utilities/steam.py]: Sorry, I'm a bit confused about what you mean by this, do you mean adding a try-catch to the import? I also moved the import to only the function that uses it.

```
def recommended_games(self, steam_id: str) -> dict:
        try:
            import steamspypi
        except ImportError:
            raise ImportError("steamspypi library is not installed.")
```

Comment (muntaqamahmood) [reply libs/langchain/langchain/utilities/steam.py]: Hi, thanks for reviewing our PR, can you kindly let us know if the error happened in the notebook?

**Review by hwchase17 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**